src/services/UserService.js

```diff
@@ -1319,24 +1319,56 @@ define(['structures/SessionCreateStruct', 'structures/UserCreateStruct', 'struct
      * @method refreshSession
      * @param {String} sessionId the session identifier (e.g. "o7i8r1sapfc9r84ae53bgq8gp4")
      * @param {Function} callback
+     * @param {Number} timeout Optional timeout in milliseconds, default is 180000 (180sec, as in 3 minutes)
+     * @param {Function} reTryCallback Optional callback to get notified if code is retrying session refresh
+     *                   NOTE: gets called every time, except on the first try!
      */
-    UserService.prototype.refreshSession = function (sessionId, callback) {
-        var that = this;
+    UserService.prototype.refreshSession = function (sessionId, callback, timeout, reTryCallback) {
+        var that = this, firstTry = true, date = new Date(), refreshSessionInfo, sessionRefresh = function(e, result) {
+            // We re-try session refresh for 180 seconds on timeouts/server errors as it is often the requests holding
+            // others back(see ConnectionManager), and is thus the one request we typically need to re-try a few
+            // times if backend has temporary downtime or is undergoing brief maintenance.
+            if (date - (new Date()) >= (timeout || 180000)) {
```

Comment:
Should be `(new Date()) - date` (other way around).

Reply:
Indeed 😅